core/auth/src/main/java/software/amazon/awssdk/auth/signer/AwsSignerExecutionAttribute.java

```diff
@@ -270,16 +272,24 @@ private static RegionScope signingRegionScopeReadMapping(SelectedAuthScheme<?> a
         if (authScheme == null) {
             return null;
         }
-        String regionScope = authScheme.authSchemeOption().signerProperty(AwsV4aHttpSigner.REGION_NAME);
-        if (regionScope == null) {
+        Set<software.amazon.awssdk.http.auth.aws.RegionScope> regionSet = authScheme.authSchemeOption().signerProperty(AwsV4aHttpSigner.REGION_SET);
+        if (regionSet == null || regionSet.isEmpty()) {
             return null;
         }
-        return RegionScope.create(regionScope);
+
+        if (regionSet.size() > 1) {
+            throw new UnsupportedOperationException("Only one RegionScope is supported when mapping to the derived attribute.");
+        }
+
+        return RegionScope.create(regionSet.stream().findFirst().get().id());
     }
 
     private static <T extends Identity> SelectedAuthScheme<?> signingRegionScopeWriteMapping(SelectedAuthScheme<T> authScheme,
                                                                                              RegionScope regionScope) {
-        String regionScopeString = regionScope == null ? null : regionScope.id();
+        Set<software.amazon.awssdk.http.auth.aws.RegionScope> regionSet =
+            regionScope != null ?
+            Collections.singleton(software.amazon.awssdk.http.auth.aws.RegionScope.create(regionScope.id())) :
```

Comment:
Unfortunately, `RegionScope`is already imported, so the FQN has to be used in these places.